ta/trusted_keys/entry.c

```diff
@@ -42,7 +44,15 @@ static TEE_Result get_random(uint32_t types, TEE_Param params[TEE_NUM_PARAMS])
 	if (!params[0].memref.buffer || !params[0].memref.size)
 		return TEE_ERROR_BAD_PARAMETERS;
 
-	TEE_GenerateRandom(params[0].memref.buffer, params[0].memref.size);
+	rng_buf = TEE_Malloc(params[0].memref.size, TEE_MALLOC_FILL_ZERO);
```

Comment:
Should the buffer be tested first? In case the caller is a TA or a PTA, the additional allocation may be unnecessary.

Reply:
Current client for this TA is Linux kernel only. IMO, at this point extra checking would be redundant.

Reply:
OK makes sense.